randovania/games/cave_story/patcher/caver_patcher.py

```diff
@@ -0,0 +1,211 @@
+                
+            mapname = logbook_node.extra.get("event_map", game_description.world_list.nodes_to_area(logbook_node).extra["map_name"])
+            event = logbook_node.extra["event"]
+            
+            if hints.get(mapname) is None:
+                hints[mapname] = {}
+            hints[mapname][event] = {
+                "text": hints_for_asset[logbook_node.string_asset_id],
+                "facepic": logbook_node.extra.get("facepic", "0000"),
+                "ending": logbook_node.extra.get("ending", "<END")
+            }
+
+        mapnames = pickups.keys() | music.keys() | entrances.keys()
+        maps = {mapname: {
+            "pickups": pickups.get(mapname, {}),
+            "music": music.get(mapname, {}),
+            "entrances": entrances.get(mapname, {}),
+            "hints": hints.get(mapname, {}),
+        } for mapname in mapnames}
+
+        # objective flags
+        starting_script = configuration.objective.script
+        # B2 falling blocks disable flag
+        if configuration.no_blocks:
+            starting_script += "<FL+1351"
+        # rocket skip enabled
+        if configuration.trick_level.level_for_trick(db.get_by_type_and_index(ResourceType.TRICK, "Dboost")).as_number >= 4:
+            starting_script += "<FL+6400"
+        # initialize HP counter
+        starting_script += set_flag(4011, configuration.starting_hp, bits=6)
+        # Camp and Labyrinth B CMP mapflags
+        starting_script += "<MP+0040<MP+0043"
+        # Softlock prevention mapflags
+        starting_script += "<MP+0032<MP+0033<MP+0036"
+
+        # Starting HP
+        if configuration.starting_hp != 3:
+            starting_script += f"<ML+{num_to_tsc_value(configuration.starting_hp - 3).decode('utf-8')}"
+
+        # TODO: starting items
+
+        # TODO: allow any starting location
+
+        # starting location flag and EVE/TRA
+        if patches.starting_location.area_name == "Start Point":
+            starting_script += "<FL+6200<EVE0091"
+        else:
+            # flags set during first cave in normal gameplay
+            starting_script += "<FL+0301<FL+0302<FL+1641<FL+1642<FL+0320<FL+0321"
+            if patches.starting_location.area_name == "Arthur's House":
+                starting_script += "<FL+6201<TRA0001:0094:0008:0004"
+            elif patches.starting_location.area_name == "Camp":
+                starting_script += "<FL+6202<TRA0040:0094:0014:0009"
+        
+        maps["Start"]["pickups"]["0201"] = starting_script
+
+        return {
+            "maps": maps,
+            "mychar": cosmetic_patches.mychar.mychar_bmp(mychar_rng),
+            "hash": get_ingame_hash(description._shareable_hash_bytes)
+        }
+    
+    def patch_game(self, input_file: Optional[Path], output_file: Path, patch_data: dict, internal_copies_path: Path, progress_update: ProgressUpdateCallable):
+        self._busy = True
+        caver_patcher.patch_files(patch_data, output_file, progress_update)
```

Comment:
might want to write the patch_data to a file for easier debugging for now.

Reply:
:+1: